storagemarket/impl/providerstates/provider_fsm.go

```diff
@@ -261,3 +261,13 @@ var ProviderFinalityStates = []fsm.StateKey{
 	storagemarket.StorageDealSlashed,
 	storagemarket.StorageDealExpired,
 }
+
+// StatesKnownBySealingSubsystem are the states on the happy path after hand-off to
+// the sealing subsystem
+var StatesKnownBySealingSubsystem = []fsm.StateKey{
```

Comment:
What is this used for? I can't find it anywhere in the code? I feel like if it's used in the markets module, having a public var a deeply nested section is maybe not the best idea?

Reply:
It's used when registering deals with the DAG store - we only want to register deals that have been handed off to the sealing subsystem (not deals that haven't yet reached that stage, or that have reached a terminal state).
I think it makes sense to have it close to the FSM, so that if more states are added to the FSM it's more likely to be noticed that they need to be added to this array as well.

Reply:
Agree with @dirkmc, we do want to keep it close to the authority in states (go-fil-markets).